File: string/first-letter-to-appear-twice/first-letter-to-appear-twice.go
package firstlettertoappeartwice

func repeatedCharacter(s string) byte {
	lookup := make(map[rune]bool)
	for i, c := range s {
		if lookup[c] {
			return s[i]
		} else {
			lookup[c] = true
		}
	}

	return s[0]
}

File: string/first-letter-to-appear-twice/first-letter-to-appear-twice.ts
function repeatedCharacter(s: string): string {
    const lookup: { [key: string]: boolean } = {};

    for (let c of s) {
        if (lookup[c] !== undefined) {
            return c;
        } else {
            lookup[c] = true;
        }
    }

    return s;
};
